tests/integration/components/bs-popover-test.js

```diff
@@ -260,6 +260,94 @@ module('Integration | Component | bs-popover', function (hooks) {
     assert.dom('#wrapper .popover').exists({ count: 1 }, 'Popover exists in place.');
   });
 
+  test('it calls onShow/onShown actions when showing popover by event [fade=false]', async function (assert) {
+    let showAction = sinon.spy();
+    this.set('show', showAction);
+    let shownAction = sinon.spy();
+    this.set('shown', shownAction);
+    await render(
+      hbs`<div id="target"><BsPopover @title="Dummy" @fade={{false}} @onShow={{this.show}} @onShown={{this.shown}} /></div>`
+    );
+    await click('#target');
+    assert.ok(showAction.calledOnce, 'show action has been called');
+    assert.ok(shownAction.calledOnce, 'shown action has been called');
+  });
+
+  test('it calls onShow/onShown actions when showing popover programmatically [fade=false]', async function (assert) {
+    let showAction = sinon.spy();
+    this.set('show', showAction);
+    let shownAction = sinon.spy();
+    this.set('shown', shownAction);
+    this.set('visible', false);
+    await render(
+      hbs`<div id="target"><BsPopover @title="Dummy" @visible={{this.visible}} @fade={{false}} @onShow={{this.show}} @onShown={{this.shown}} /></div>`
+    );
+    this.set('visible', true);
+    await settled();
+    assert.ok(showAction.calledOnce, 'show action has been called');
+    assert.ok(shownAction.calledOnce, 'shown action has been called');
+  });
+
+  test('it aborts showing if onShow action returns false', async function (assert) {
+    let showAction = sinon.stub();
+    showAction.returns(false);
+    this.set('show', showAction);
+    let shownAction = sinon.spy();
+    this.set('shown', shownAction);
+    await render(
+      hbs`<div id="target"><BsPopover @title="Dummy" @fade={{false}} @onShow={{this.show}} @onShown={{this.shown}} /></div>`
+    );
+    await click('#target');
+    assert.ok(showAction.calledOnce, 'show action has been called');
+    assert.notOk(shownAction.calledOnce, 'show action has not been called');
```

Comment:
I think it would be better to assert that it hasn't been called. Otherwise assertion would also pass if it has been called more than once.

```suggestion
    assert.ok(shownAction.notCalled, 'show action has not been called');
```